glib/SafeObjectHandle.cs

```diff
@@ -34,21 +37,28 @@ class SafeObjectHandle : SafeHandleZeroOrMinusOneIsInvalid
 		{
 		}
 
-		SafeObjectHandle (IntPtr handle) : base(true)
+		protected SafeObjectHandle (IntPtr handle) : base (true)
 		{
 			SetHandle (handle);
 		}
 
+#if DEBUG_SAFEHANDLES
+		public static Func<IntPtr, SafeObjectHandle> InternalCreateHandle;
+#endif
+
 		static readonly Dictionary<IntPtr, ToggleRef> Objects = new Dictionary<IntPtr, ToggleRef> (IntPtrEqualityComparer.Instance);
 
 		public static SafeObjectHandle Create (Object obj, IntPtr handle)
 		{
 			if (handle == IntPtr.Zero)
 				return Zero;
 
-			var safeHandle = new SafeObjectHandle (handle) {
-				tref = new ToggleRef (obj, handle)
-			};
+#if DEBUG_SAFEHANDLES
+			var safeHandle = new InternalCreateHandle (handle);
```

Comment:
Shouldn't this be `var safeHandle = InternalCreateHandle?.Invoke (handle) ?? new SafeObjectHandle (handle);` ?